opal/mca/pool/base/pool_base_open.c

```diff
@@ -0,0 +1,77 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2017      Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+
+#include "opal_config.h"
+
+#include "opal/constants.h"
+#include "opal/mca/mca.h"
+#include "opal/mca/base/base.h"
+#include "opal/mca/pool/pool.h"
+#include "opal/mca/pool/base/base.h"
+
+
+/*
+ * The following file was created by configure.  It contains extern
+ * statements and the definition of an array of pointers to each
+ * component's public mca_base_component_t struct.
+ */
+#include "opal/mca/pool/base/static-components.h"
+
+/*
+ * Globals
+ */
+opal_pool_base_component_2_0_0_t *opal_pool = NULL;
+
+
+/*
+ * Function for finding and opening either all MCA components, or the one
+ * that was specifically requested via a MCA parameter.
+ */
+static int opal_pool_base_open(mca_base_open_flag_t flags)
```

Comment:
Why did this require a custom component open routine?

Reply:
i did quite some copy/paste from existing frameworks, so this could be unnecessary. i will double check that.